.github/workflows/build.yml

```diff
@@ -1,33 +1,37 @@
 name: Build & Test
 
-on: [push, pull_request]
+on: [push, pull_request, workflow_dispatch]
 
 jobs:
   build:
-
+    name: Build
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Prepare cache key
+        id: cache-key
+        shell: bash
+        run: echo "::set-output name=date::$(/bin/date -u "+%Y-%m")"
       - name: Cache Maven dependencies
-        uses: actions/cache@v2.1.6
+        uses: actions/cache@v2
         with:
           path: ~/.m2/repository
-          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: |
-            ${{ runner.os }}-maven-
-      - name: Cache frontend dependencies
-        uses: actions/cache@v2.1.6
+          # The cache key is changed every month to prevent unlimited growth.
+          key: maven-cache-${{ steps.cache-key.outputs.date }}
```

Comment:
The cache we used before was never cleaned and kept growing build after build (there's a 5 GB limit for each cache though). Now it will be cleaned each month and the planet will be a bit happier 😄

Reply:
not only the planet :D